test/compiler.test.js

```diff
@@ -1014,6 +1014,201 @@ describe('compiler', function() {
       expect(instructions.config).to.not.have.property('modified');
     });
+
+    describe('for mixins', function() {
+      it('supports `mixinSources` option', function() {
+        appdir.writeFileSync('mixins/other.js', '');
+
+        var instructions = boot.compile({
+          appRootDir: appdir.PATH,
+          mixinSources: ['mixins'],
+          normalization: 'classify'
+        });
+
+        expect(instructions.mixins).to.have.property('Other');
+      });
+
+      it('resolves relative path in `mixinSources` option', function() {
+        appdir.writeFileSync('custom-mixins/vehicle.js', '');
+
+        var instructions = boot.compile({
+          appRootDir: appdir.PATH,
+          mixinSources: ['./custom-mixins']
+        });
+
+        expect(instructions.mixins).to.have.property('Vehicle');
+      });
+
+      it('resolves module relative path in `mixinSources` option', function() {
+        appdir.writeFileSync('node_modules/custom-mixins/vehicle.js', '');
+
+        var instructions = boot.compile({
+          appRootDir: appdir.PATH,
+          mixinSources: ['custom-mixins']
+        });
+
+        expect(instructions.mixins).to.have.property('Vehicle');
+      });
+
+      it('supports `mixins` option in model config', function() {
+        appdir.createConfigFilesSync({}, {}, {
+          _meta: {
+            mixins: ['mixins']
+          }
+        });
+
+        appdir.writeFileSync('mixins/other.js', '');
+        var instructions = boot.compile(appdir.PATH);
+
+        expect(instructions.mixins).to.have.property('Other');
+      });
+
+      it('resolves relative path in `mixins` option', function() {
+        appdir.createConfigFilesSync({}, {}, {
+          _meta: {
+            mixins: ['./custom-mixins']
+          }
+        });
+
+        appdir.writeFileSync('custom-mixins/vehicle.js', '');
+        var instructions = boot.compile(appdir.PATH);
+
+        expect(instructions.mixins).to.have.property('Vehicle');
+      });
+
+      it('resolves module relative path in `mixins` option', function() {
+        appdir.createConfigFilesSync({}, {}, {
+          _meta: {
+            mixins: ['custom-mixins']
+          }
+        });
+
+        appdir.writeFileSync('node_modules/custom-mixins/vehicle.js', '');
+        var instructions = boot.compile(appdir.PATH);
+
+        expect(instructions.mixins).to.have.property('Vehicle');
+      });
+
+      it('picks mixins from `mixinSources` when provided', function() {
+        appdir.writeFileSync('custom-mixins/other.js', '');
+        appdir.writeFileSync('mixins/default.js', '');
+
+        var instructions = boot.compile({
+          appRootDir: appdir.PATH,
+          mixinSources: ['custom-mixins']
+        });
+
+        expect(instructions.mixins).to.have.property('Other');
+        expect(instructions.mixins).to.not.have.property('Default');
+      });
+
+      it('supports mixin name normalization - classify', function() {
```

Comment:
Please wrap all tests called "supports mixin name normalization" in a new "describe" block and add a beforeEach hook to create the mixin files. 

We should probably test file names containing a space too.

```js
describe('name normalization', function() {
  beforeEach(function() {
    appdir.writeFileSync('mixins/foo.js', '');
    appdir.writeFileSync('mixins/time-stamps.js', '');
    appdir.writeFileSync('mixins/camelCase.js', '');
    appdir.writeFileSync('mixins/PascalCase.js', '');
    appdir.writeFileSync('mixins/space name.js', '');
  });

  it('supports "classify"', function() {
    var instructions = //;
    expect(_.keys(instructions.mixins)).to.eql([
      'Foo', 'TimeStamps', 'CamelCase', 'PascalCase', 'SpaceName'
    ]);
  });
  // etc.
});
```